File: 0226-invert-binary-tree/0226-invert-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    
    return invertOrder(root)
    
    function invertOrder(root){
        if(!root) return null
    
        let left=invertOrder(root.left)
        let right=invertOrder(root.right)
    
        root.right = left;
        root.left = right;
    
        return root;
    }
};

File: 0226-invert-binary-tree/0226-invert-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def invertTree(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        self.invertOrder(root)
        return root

    def invertOrder(self,root):
        if not root:
            return

        self.invertOrder(root.left)
        self.invertOrder(root.right)
        self.swap(root)

    def swap(self,root):
        if not root:
            return
        temp = root.left
        root.left = root.right
        root.right = temp
        